game-core/src/test/java/games/strategy/triplea/delegate/battle/casualty/CasualtySortingUtilTest.java

```diff
@@ -0,0 +1,159 @@
+
+    assertThat(units, is(List.of(unitA, unitB)));
+  }
+
+  @Test
+  void sortByMovement() {
+    final UnitType unitType1 = mock(UnitType.class);
+    when(unitType1.getName()).thenReturn("A");
+    final UnitAttachment unitAttachment1 = mock(UnitAttachment.class);
+    when(unitType1.getAttachment(UNIT_ATTACHMENT_NAME)).thenReturn(unitAttachment1);
+    when(unitAttachment1.getMovement(player)).thenReturn(2);
+    final Unit unitWith2Movement = new Unit(unitType1, player, gameData);
+
+    final UnitType unitType2 = mock(UnitType.class);
+    when(unitType2.getName()).thenReturn("A");
+    final UnitAttachment unitAttachment2 = mock(UnitAttachment.class);
+    when(unitAttachment2.getMovement(player)).thenReturn(1);
+    when(unitType2.getAttachment(UNIT_ATTACHMENT_NAME)).thenReturn(unitAttachment2);
+    final Unit unitWith1Movement = new Unit(unitType2, player, gameData);
+
+    final List<Unit> units = Arrays.asList(unitWith2Movement, unitWith1Movement);
+
+    CasualtySortingUtil.sortPreBattle(units);
+
+    assertThat(units, is(List.of(unitWith1Movement, unitWith2Movement)));
+  }
+
+  @Test
+  void sortByMarine() {
+    final UnitType unitType1 = mock(UnitType.class);
+    when(unitType1.getName()).thenReturn("A");
+    final UnitAttachment unitAttachment1 = mock(UnitAttachment.class);
+    when(unitType1.getAttachment(UNIT_ATTACHMENT_NAME)).thenReturn(unitAttachment1);
+    lenient().when(unitAttachment1.getMovement(player)).thenReturn(1);
+    when(unitAttachment1.getIsMarine()).thenReturn(0);
+    final Unit unitWithNoMarine = spy(new Unit(unitType1, player, gameData));
+    lenient().when(unitWithNoMarine.getWasAmphibious()).thenReturn(true);
+
+    final UnitType unitType2 = mock(UnitType.class);
+    when(unitType2.getName()).thenReturn("A");
+    final UnitAttachment unitAttachment2 = mock(UnitAttachment.class);
+    lenient().when(unitAttachment2.getMovement(player)).thenReturn(1);
+    when(unitAttachment2.getIsMarine()).thenReturn(1);
+    when(unitType2.getAttachment(UNIT_ATTACHMENT_NAME)).thenReturn(unitAttachment2);
+    final Unit unitWith1Marine = spy(new Unit(unitType2, player, gameData));
+    when(unitWith1Marine.getWasAmphibious()).thenReturn(true);
+
+    final UnitType unitType3 = mock(UnitType.class);
+    when(unitType3.getName()).thenReturn("A");
+    final UnitAttachment unitAttachment3 = mock(UnitAttachment.class);
+    lenient().when(unitAttachment3.getMovement(player)).thenReturn(1);
+    when(unitAttachment3.getIsMarine()).thenReturn(2);
+    when(unitType3.getAttachment(UNIT_ATTACHMENT_NAME)).thenReturn(unitAttachment3);
+    final Unit unitWith2Marine = spy(new Unit(unitType3, player, gameData));
+    when(unitWith2Marine.getWasAmphibious()).thenReturn(true);
+
+    final List<Unit> units = Arrays.asList(unitWith2Marine, unitWith1Marine, unitWithNoMarine);
+
+    CasualtySortingUtil.sortPreBattle(units);
+
+    assertThat(units, is(List.of(unitWithNoMarine, unitWith1Marine, unitWith2Marine)));
```

Comment:
nit, have you considered using the collection matcher? IE: `IsCollectionContaining`